ios/testmanagerd/xcuitestrunner.go

```diff
@@ -389,40 +338,70 @@ func runXUITestWithBundleIdsXcode15Ctx(
 		"ubiquitous test identifiers":              uint64(1),
 	}}
 
-	ideDaemonProxy2 := newDtxProxyWithConfig(conn2, testconfig)
+	ideDaemonProxy2 := newDtxProxyWithConfig(conn2, testconfig, testListener)
 	caps, err := ideDaemonProxy1.daemonConnection.initiateControlSessionWithCapabilities(localCaps)
 	if err != nil {
-		return err
+		return fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot initiate a control session with capabilities: %w", err)
 	}
 	log.WithField("caps", caps).Info("got capabilities")
 	authorized, err := ideDaemonProxy2.daemonConnection.authorizeTestSessionWithProcessID(pid)
 	if err != nil {
-		return err
+		return fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot authorize test session: %w", err)
 	}
 	log.WithField("authorized", authorized).Info("authorized")
 
 	err = ideDaemonProxy2.daemonConnection.initiateControlSession(pid, proto)
 	if err != nil {
-		return err
+		return fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot initiate a control session: %w", err)
 	}
 	log.Debug("control session initiated")
 
 	ideInterfaceChannel := ideDaemonProxy1.dtxConnection.ForChannelRequest(ProxyDispatcher{id: "dtxproxy:XCTestDriverInterface:XCTestManager_IDEInterface"})
 
 	err = ideDaemonProxy1.daemonConnection.startExecutingTestPlanWithProtocolVersion(ideInterfaceChannel, proto)
 
-	time.Sleep(60 * time.Second)
+	if ctx != nil {
+		select {
+		case <-ctx.Done():
+			log.Infof("Killing test runner with pid %d ...", pid)
+			err = appserviceConn.KillProcess(int(pid))
+			if err != nil {
+				return fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot kill test process: %w", err)
+			}
+			log.Info("Test runner killed with success")
+
+			if testListener != nil {
+				(*testListener).TestRunnerKilled()
+			}
+		}
+		return nil
+	}
+
+	<-ideDaemonProxy1.proxyDispatcher.closeChannel
+	log.Debugf("Killing test runner with pid %d ...", pid)
+	err = appserviceConn.KillProcess(int(pid))
+	if err != nil {
+		return fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot kill test process: %w", err)
+	}
+	log.Debugf("Test runner killed with success")
+	if testListener != nil {
+		(*testListener).TestRunnerKilled()
```

Comment:
What `TestRunnerKilled` should signal is if the process died unexpectedly, but that's not the case here, and as we don't observe the process, we can remove this completely.

Reply:
Removed.